apps/server/src/modules/school/domain/testing/school-year.factory.ts

```diff
@@ -0,0 +1,27 @@
+import { BaseFactory } from '@shared/testing';
+import { ObjectId } from 'bson';
+import { SchoolYear, SchoolYearProps } from '../do';
+
+type SchoolYearTransientParams = {
+	startYear: number;
+};
+
+class SchoolYearFactory extends BaseFactory<SchoolYear, SchoolYearProps, SchoolYearTransientParams> {
+	public withStartYear(startYear: number): this {
+		this.rewindSequence();
+		return this.transient({ startYear });
+	}
+}
+
+export const schoolYearFactory = SchoolYearFactory.define(SchoolYear, ({ transientParams, sequence }) => {
+	const id = new ObjectId().toHexString();
+
+	const startYearWithoutSequence = transientParams?.startYear ?? new Date().getFullYear();
```

Comment:
Ich finde das Verhalten von ?? gegenüber dem || immer etwas merkwürdig. 
Bei defined, undefined, null und true verhält es sich identisch. Bei false nimmt es false als Resultat an. 
false ?? 3 -> false
false || 3 -> 3

Reply:
Der Unterschied wird hier erklärt: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing#description

An der Stelle würds z.B. einen Unterschied machen, wenn `startYear` 0 wäre, weil 0 falsy ist. Ist unwahrscheinlich, aber es soll ja ein null-Check sein und das drückt ?? aus.

Reply:
Die Arguementation von dir verstehe ich nicht. Das ?? verwendet man wenn man 0, '', or NaN als valid Werte erwartet. Wenn das keine validen Werte sind würde man || verwenden.  Ich würde aber behaupten die Werte sind für ein Jahr definitiv keine validen Werte.

"This behavior may cause unexpected consequences if you consider 0, '', or NaN as valid values."